src/coreclr/src/jit/lsraxarch.cpp

```diff
@@ -904,6 +903,121 @@ int LinearScan::BuildRMWUses(GenTreeOp* node, regMaskTP candidates)
     return srcCount;
 }
 
+#ifdef FEATURE_SIMD
+int LinearScan::BuildRMWUsesSIMDUnary(GenTreeSIMD* node)
```

Comment:
Yes but I first want to double check if this is truly needed. This was extracted from the existing `BuildRMWUses` and at least the x86 byte code is suspicious, I don't think it applies to SIMD.

Reply:
The x86 code is definitely useless (and broken if ever used on a SIMD node). It's intended for byte instructions like `add al, bl`. I suspect that the setting of `tgtPrefUse` is also pointless, that's probably useful for instructions like `inc eax`.

Reply:
Agreed that it's unlikely to be useful for SIMD, though I haven't checked for certain that there are no RMW unary SIMD ops.

Reply:
OK, the RMW stuff is pretty convoluted. I tried to clean it up a bit more but ended up leaving it alone since it's not clear if the effort is worthwhile. Ultimately HW intrinsic do the same thing and do it better, so it makes more sense to spend time replacing SIMD with HWINTRINSIC instead of attempting to improve SIMD.

So I added function headers and extra comments explaining some of the issues I noticed while looking at this.